columnflow/tasks/framework/base.py

```diff
@@ -115,23 +119,23 @@ def req_params(cls, inst, **kwargs):
         other class. The parameters are used when calling ``Task.req(self)``.
         """
         # always prefer certain parameters given as task family parameters (--TaskFamily-parameter)
-        _prefer_cli = set(law.util.make_list(kwargs.get("_prefer_cli", [])))
-        _prefer_cli |= {
+        _prefer_cli = law.util.make_set(kwargs.get("_prefer_cli", [])) | {
             "version", "workflow", "job_workers", "poll_interval", "walltime", "max_runtime",
             "retries", "acceptance", "tolerance", "parallel_jobs", "shuffle_jobs", "htcondor_cpus",
             "htcondor_gpus", "htcondor_pool",
         }
         kwargs["_prefer_cli"] = _prefer_cli
 
-        # when cls accepts a version, but non was actively requested, use the version map to assign it
+        # when cls accepts a version but none was requested use the version map to assign it
         version_map = None
         if isinstance(getattr(cls, "version", None), luigi.Parameter) and "version" not in kwargs:
             version_map = cls.get_version_map(inst)
```

Comment:
Unrelated to the PR, but since I'm seeing this right now:, a few days ago, I tried running some tasks using the `config_inst.x.versions` aux instead of adding a version manually, but when running a task without specifying a version, the task would crash.

Reply:
The version map lookup only comes into play when a `req()` is called, and for that the triggered task still needs a version to be set.